### ISSUE TITLE
Seg fault in `SHA256_Final` with  asm and`-fno-omit-frame-pointer` in linux-ppc64le

### ISSUE THREAD

**Issue by shigeki** [openssl/openssl#7584]
I'm now working on upgrading OpenSSL-1.1.1 in Node.js. 
After upgrading to 1.1.1, digesting of SHA256 in linux-ppc64le was crashed with a segmentation fault due to a build option of `-fno-omit-frame-pointer` which is added by Node.js build configurations. 

The details are shown as blow for reproduction in bare OpenSSL-1.1.1 sources.
```
# uname -a
Linux xxxxl-ubuntu1404-ppc64--le-2 3.13.0-153-generic #203-Ubuntu SMP Thu Jun 14 08:52:31 UTC 2018 ppc64le ppc64le ppc64le GNU/Linux

# gcc-4.9  --version
gcc-4.9 (Ubuntu 4.9.4-2ubuntu1~14.04.1) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

# gcc-4.9 -Wa,-v -c -o /dev/null -x assembler /dev/null 2>&1
GNU assembler version 2.26.1 (powerpc64le-linux-gnu) using BFD version (GNU Binutils for Ubuntu) 2.26.1

# ./config -fno-omit-frame-pointer no-shared
Operating system: ppc64le-whatever-linux2
Configuring OpenSSL version 1.1.1 (0x1010100fL) for linux-ppc64le
..

# make
..
CC="gcc-4.9" /usr/bin/perl crypto/sha/asm/sha512p8-ppc.pl linux64le crypto/sha/sha256p8-ppc.s
gcc-4.9 -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -fno-omit-frame-pointer -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_BN_ASM_MONT -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DAES_ASM -DVPAES_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib/engines-1.1\"" -DNDEBUG  -c -o crypto/sha/sha256p8-ppc.o crypto/sha/sha256p8-ppc.s
..

# echo 'hi' | ./apps/openssl dgst -sha256
Segmentation fault (core dumped)

# ../gdb-8.2/gdb/gdb ./apps/openssl core
GNU gdb (GDB) 8.2
..
Using host libthread_db library "/lib/powerpc64le-linux-gnu/libthread_db.so.1".
Core was generated by `./apps/openssl dgst -sha256 '.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000000101f9bf8 in SHA256_Final ()
(gdb)
```
My bisecting shows that  https://github.com/openssl/openssl/pull/6406/commits/3c1ece1f59bcf32f477860e8f517214c4ffc7dca with `-fno-omit-frame-pointer` cause the issue. But building them with `--debug` or `no-asm` are working well.

I'm not familiar with PPC asm. Do we have to avoid  using`-fno-omit-frame-pointer` with asm in linux-ppc64le? 

Ref: https://github.com/nodejs/node/issues/18770

**Comment by sam-github** [comment]
@dot-asm is https://github.com/openssl/openssl/commit/3c1ece1f59bcf32f477860e8f517214c4ffc7dca supposed to work when the frame pointer cannot be omitted?

**Comment by dot-asm** [comment]
> is 3c1ece1 supposed to work when the frame pointer cannot be omitted?

It's not about supposed to work, but about supposed to not have typos. Addressed in #7643.

**Comment by shigeki** [comment]
Fixed by https://github.com/openssl/openssl/pull/7643. Thanks.
